assets/js/blocks/cart-checkout/checkout/form/index.js

```diff
@@ -0,0 +1,323 @@
+						'Enter the address that matches your card or payment method.',
+						'woo-gutenberg-products-block'
+					) }
+				>
+					<AddressForm
+						id="billing"
+						onChange={ setBillingFields }
+						type="billing"
+						values={ billingFields }
+						fields={ Object.keys( defaultAddressFields ) }
+						fieldConfig={ addressFieldsConfig }
+					/>
+				</FormStep>
+			) }
+			{ needsShipping && (
+				<FormStep
+					id="shipping-option"
+					disabled={ checkoutIsProcessing }
+					className="wc-block-checkout__shipping-option"
+					title={ __(
+						'Shipping options',
+						'woo-gutenberg-products-block'
+					) }
+					description={
+						getShippingRatesRateCount( shippingRates ) > 1
+							? __(
+									'Select shipping options below.',
+									'woo-gutenberg-products-block'
+							  )
+							: ''
+					}
+				>
+					{ isEditor &&
+					! getShippingRatesPackageCount( shippingRates ) ? (
+						<NoShippingPlaceholder />
+					) : (
+						<ShippingRatesControl
+							noResultsMessage={ __(
+								'There are no shipping options available. Please ensure that your address has been entered correctly, or contact us if you need any help.',
+								'woo-gutenberg-products-block'
+							) }
+							renderOption={ renderShippingRatesControlOption }
+							shippingRates={ shippingRates }
+							shippingRatesLoading={ shippingRatesLoading }
+						/>
+					) }
+				</FormStep>
+			) }
+			{ cartNeedsPayment && (
+				<FormStep
+					id="payment-method"
+					disabled={ checkoutIsProcessing }
+					className="wc-block-checkout__payment-method"
+					title={ __(
+						'Payment method',
+						'woo-gutenberg-products-block'
+					) }
+					description={
+						Object.keys( paymentMethods ).length > 1
+							? __(
+									'Select a payment method below.',
+									'woo-gutenberg-products-block'
+							  )
+							: ''
+					}
+				>
+					<StoreNoticesProvider context="wc/payment-area">
+						<PaymentMethods />
+					</StoreNoticesProvider>
+				</FormStep>
+			) }
+			{ showOrderNotes && (
+				<FormStep id="order-notes" showStepNumber={ false }>
+					<CheckoutOrderNotes
+						disabled={ checkoutIsProcessing }
+						onChange={ setOrderNotes }
+						placeholder={
+							needsShipping
+								? __(
+										'Notes about your order, e.g. special notes for delivery.',
+										'woo-gutenberg-products-block'
+								  )
+								: __(
+										'Notes about your order.',
+										'woo-gutenberg-products-block'
+								  )
+						}
+						value={ orderNotes }
+					/>
+				</FormStep>
+			) }
+		</Form>
+	);
+};
+
+CheckoutForm.propTypes = {
+	loginToCheckoutUrl: PropTypes.string.isRequired,
+	requireCompanyField: PropTypes.bool.isRequired,
+	requirePhoneField: PropTypes.bool.isRequired,
+	showApartmentField: PropTypes.bool.isRequired,
+	showCompanyField: PropTypes.bool.isRequired,
+	showOrderNotes: PropTypes.bool.isRequired,
+	showPhoneField: PropTypes.bool.isRequired,
+};
+
+export default CheckoutForm;
```

Comment:
I see this is the business end, still a big component. Good to start breaking it down, and we can iterate more in future 😄

Reply:
Yeah, at some point it might make sense to keep splitting this one into smaller chunks. But I feel now it's much more self-contained (it's only the form) than before (it was the entire block inside a single component).